<?xml version="1.0"?>
<launch>

  <arg name="input_pointcloud_topic" default="/arm_cam3d/depth_registered/points" />
  <arg name="target_frame" default="base_link" />
  <arg name="scene_segmentation_config_file" default="$(find mir_object_segmentation)/ros/config/scene_segmentation_constraints.yaml" />

  <group ns="mir_perception">
    <rosparam file="$(arg scene_segmentation_config_file)" command="load"/>
    <node pkg="mir_object_segmentation" type="scene_segmentation_node" name="scene_segmentation" output="screen">
      <remap from="~input" to="$(arg input_pointcloud_topic)" />
      <remap from="~output/object_list" to="/mir_perception/scene_segmentation/output/object_list"/>
      <param name="target_frame_id" value="$(arg target_frame)" type="str" />
      <param name="logdir" value="/tmp/" />
    </node>
  </group>

</launch>
